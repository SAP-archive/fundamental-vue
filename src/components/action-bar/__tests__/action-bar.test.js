import { createLocalVue, mount } from "@vue/test-utils";
import FundamentalVue from "../../../index";

const mountActionBar = template => {
  const localVue = createLocalVue();
  localVue.use(FundamentalVue);
  return mount({ template }, { localVue });
};

describe("ActionBar", () => {
  it("renders back-button-slot", () => {
    const wrapper = mountActionBar(`
    <fd-action-bar>
      <template #back-button>
        <fd-button styling="light" icon="nav-back" />
      </template>
    </fd-action-bar>
  `);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders header-title-slot", () => {
    const wrapper = mountActionBar(`
    <fd-action-bar>
      <template #header-title>
        <fd-action-bar-title>Action Bar Title</fd-action-bar-title>
      </template>
    </fd-action-bar>
  `);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly in it's full glory", () => {
    const wrapper = mountActionBar(`
      <fd-action-bar>
        <template #back>
          <fd-action-bar-back>
            <fd-button styling="light" icon="nav-back" />
          </fd-action-bar-back>
        </template>
        <template #header>
          <fd-action-bar-header>
            <template #title>
              <fd-action-bar-title>Action Bar Title</fd-action-bar-title>
            </template>
            <template #description>
              <fd-action-bar-description>Action Bar Description</fd-action-bar-description>
            </template>
          </fd-action-bar-header>
        </template>
        <template #actions>
          <fd-action-bar-actions>
            <fd-button type="positive" icon="accept">Approve</fd-button>
            <fd-button type="negative" icon="decline">Reject</fd-button>
          </fd-action-bar-actions>
        </template>
      </fd-action-bar>
    `);
    expect(wrapper.element).toMatchSnapshot();
  });
});
