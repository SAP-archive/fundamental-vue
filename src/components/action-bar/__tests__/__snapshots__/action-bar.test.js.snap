// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar renders back-button-slot 1`] = `
<div
  class="fd-action-bar"
>
  <div
    class="fd-action-bar__back"
  >
    <button
      class="fd-button--light sap-icon--nav-back"
    />
  </div>
   
  <div
    class="fd-action-bar__header"
  >
     
  </div>
   
</div>
`;

exports[`ActionBar renders correctly in it's full glory 1`] = `
<div
  class="fd-action-bar"
>
  <div
    class="fd-action-bar__back"
  >
    <button
      class="fd-button--light sap-icon--nav-back"
    />
  </div>
   
  <div
    class="fd-action-bar__header"
  >
    <h3
      class="fd-action-bar__title"
    >
      Action Bar Title
    </h3>
     
    <p
      class="fd-action-bar__description"
    >
      Action Bar Description
    </p>
  </div>
   
  <div
    class="fd-action-bar__actions"
  >
    <button
      class="fd-button fd-button--positive sap-icon--accept"
    >
      Approve
    </button>
     
    <button
      class="fd-button fd-button--negative sap-icon--decline"
    >
      Reject
    </button>
  </div>
</div>
`;

exports[`ActionBar renders header-title-slot 1`] = `
<div
  class="fd-action-bar"
>
  <div
    class="fd-action-bar__back"
  />
   
  <div
    class="fd-action-bar__header"
  >
    <h3
      class="fd-action-bar__title"
    >
      Action Bar Title
    </h3>
     
  </div>
   
</div>
`;
